### PR TITLE
Firestore: Add ability to configure long polling timeout

### PR DESCRIPTION
**Pull request by dconeybe**
Add a new setting for Firestore: `idleHttpRequestTimeoutSeconds`. This new option can be used to configure the SDK’s underlying network transport (WebChannel) when _long-polling_ is used, such as when `experimentalAutoDetectLongPolling` or `experimentalForceLongPolling` is set to `true`.

For convenience, here is the documentation for the new `idleHttpRequestTimeoutSeconds` property, copied from this PR's code:
> The desired maximum timeout interval, in seconds, to complete a long-polling GET response. Valid values are integers between 5 and 30, inclusive.
>
> By default, when long-polling is used the "hanging GET" request sent by the client times out after 30 seconds. To request a different timeout from the server, set this setting with the desired timeout.
>
> Changing the default timeout may be useful, for example, if the buffering proxy that necessitated enabling long-polling in the first place has a shorter timeout for hanging GET requests, in which case setting the long-polling timeout to a shorter value, such as 25 seconds, may fix prematurely-closed hanging GET requests.

Below is an example that changes the long-polling "hanging get" timeout from the default (30 seconds) to 25 seconds:
```ts
const db = initializeFirestore(app, {
  experimentalAutoDetectLongPolling: true,
  experimentalLongPollingOptions: {
    idleHttpRequestTimeoutSeconds: 25
  }
});
```

Googlers see b/266868871

Fixes: #6987

### CODE REVIEW

**Review by ehsannas — APPROVED** [review]
Tom was right! ;-P

File: packages/firestore/src/lite-api/settings.ts [hunk 1]
```diff
@@ -29,6 +34,18 @@ import { validateIsNotUsedTogether } from '../util/input_validation';
 export const DEFAULT_HOST = 'firestore.googleapis.com';
 export const DEFAULT_SSL = true;
 
+// The minimum long-polling timeout is hardcoded on the server. The value here
+// should be kept in sync with the value used by the server, as the server will
+// silently ignore a value below the minimum and fall back to the default.
+// Googlers see http://google3/net/webchannel/internal/webchannel_config.cc;l=118;rcl=510899643
```
[the remark this answers is not in the document]
Comment (dconeybe) [reply packages/firestore/src/lite-api/settings.ts]: Good point. Done.

**Review by markarndt — APPROVED** [review]

**Review by hsubox76 — COMMENTED** [review]
Can you add a changeset?

**Comment by dconeybe** [comment]
> Can you add a changeset?

Done. Thanks for pointing that out.

**Review by hsubox76 — APPROVED** [review]

**Pull request merged**